zipkin-collector/activemq/src/main/java/zipkin2/collector/activemq/ActiveMQCollector.java

```diff
@@ -0,0 +1,233 @@
+    }
+
+    @Override
+    public Builder metrics(CollectorMetrics metrics) {
+      if (metrics == null) {
+        throw new NullPointerException("metrics == null");
+      }
+      this.delegate.metrics(this.metrics);
+      return this;
+    }
+
+    public Builder addresses(String addresses) {
+      this.addresses = addresses;
+      return this;
+    }
+
+    public Builder connectionFactory(ActiveMQConnectionFactory connectionFactory) {
+      if (connectionFactory == null) {
+        throw new NullPointerException("connectionFactory == null");
+      }
+      return this;
+    }
+
+    /**
+     * Queue zipkin spans will be consumed from. Defaults to "zipkin".
+     */
+    public Builder queue(String queue) {
+      if (queue == null) {
+        throw new NullPointerException("queue == null");
+      }
+      return this;
+    }
+
+    public Builder username(String username) {
+      connectionFactory.setUserName(username);
+      return this;
+    }
+
+    /** The password to use when connecting to the broker. Defaults to "guest" */
+    public Builder password(String password) {
+      connectionFactory.setPassword(password);
+      return this;
+    }
+
+    @Override
+    public ActiveMQCollector build() {
+      return new ActiveMQCollector(this);
+    }
+  }
+
+  final String queue;
+  final LazyInit connection;
+
+  ActiveMQCollector(Builder builder) {
+    this.queue = builder.queue;
+    this.connection = new LazyInit(builder);
+  }
+
+  @Override
+  public ActiveMQCollector start() {
+    connection.get();
+    return this;
+  }
+
+  @Override
+  public CheckResult check() {
+    try {
+      CheckResult failure = connection.failure.get();
+      if (failure != null) return failure;
+      return CheckResult.OK;
+    } catch (RuntimeException e) {
+      return CheckResult.failed(e);
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    connection.close();
+  }
+
+  /** Lazy creates a connection and a queue before starting consumers */
+  static final class LazyInit  {
+    final Builder builder;
+    final AtomicReference<CheckResult> failure = new AtomicReference<>();
+    Connection connection;
+
+    LazyInit(Builder builder) {
+      this.builder = builder;
+    }
+
+    protected Connection  compute() {
+      try {
+        builder.connectionFactory.setBrokerURL("failover:("+builder.addresses+")?initialReconnectDelay=100");
+        connection = builder.connectionFactory.createConnection();
+        Session session=connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        connection.start();
+        Destination destination = session.createQueue(builder.queue);
+        MessageConsumer messageConsumer = session.createConsumer(destination);
+        Collector collector = builder.delegate.build();
+        CollectorMetrics metrics = builder.metrics;
+        messageConsumer.setMessageListener(new ActiveMQSpanConsumerMessageListener(collector,metrics));
+      }catch (Exception e){
```

Comment:
use multi-catch for what is actually thrown, unless literally Exception is thrown